frontend/src/components/pages/AddProduct.js

```diff
@@ -0,0 +1,232 @@
+import React, { useState } from "react";
+import ProductDataService from "../../client/product";
+import styled from "styled-components";
+
+const AddProduct = () => {
+    const initialProductState = {
+        title: "",
+        description: "",
+        price: "",
+        stock: "",
+        isActive: true,
+        categoryId: ""
+
+    };
+    const [product, setProduct] = useState(initialProductState);
```

Comment:
Já que esse `initialProductState` não necessita de nenhuma info passada por porps para o component, você pode criar essa variavel fora do componente. Caso contrário, toda ver que o componente for renderizado, a variável irá ser criada também, o que não é necessário.

```suggestion

const initialProductState = {
    title: "",
    description: "",
    price: "",
    stock: "",
    isActive: true,
    categoryId: ""
};

const AddProduct = () => {
    const [product, setProduct] = useState(initialProductState);
```